src/collapse/dtrace.rs

```diff
@@ -113,7 +114,7 @@ impl From<Options> for Folder {
     fn from(opt: Options) -> Self {
         Self {
             stack: VecDeque::default(),
-            occurrences: HashMap::default(),
+            occurrences: HashMap::with_capacity(100),
```

Comment:
If this indeed results in a performance increase, then we should do this for `perf` too. Also, why 100 specifically?

Reply:
Just gut feeling and experience, having something in the range of 100 elements in the stack seems reasonable. The example file has 400ish. I experimented a bit too, 1000 was significantly slower, so I picked something that seemed the right order of magnitude

Reply:
This isn't the number of things in the stack though, it's the number of distinct observed total stacks, which I'd expect to be more than 100. :man_shrugging: I'm fine with ~100 if you've found that it empirically seems to be a good balance. I'd personally lean towards slightly higher + power of two, but I don't feel too strongly about it.

Reply:
tried a few more, 512 still doesn't make a noticeable difference so we can settle there :) still not at the point where I saw degradation so it should be good.